### ISSUE TITLE
JSON Body Not parsing with body-parser

### ISSUE THREAD

**Issue by rajivharlalka** [supertokens/supertokens-node#269]
Along with the use of body-parser as a parsing agent, the body for the middleware doesnt gets parsed.

```js
// The line of code that created an issue
app.use(bodyParser.urlencoded({extended: true}));
```

SuperTokens-node version: 8.6.0
node version : 16.13.2

**Comment by MassiGy** [comment]
Hey there, 
I suggest you use the built-in express method ` express.urlencoded({extended:true}) `, which is supported in the express 4.x (I believe).
Besides that, I'am really new to open source and **I would love to help and contribute**, please assign to me some issues that needs to be fixed.

Massiles.
Backend Web Developer.

**Comment by rishabhpoddar** [comment]
hey @MassiGy from the lib's point of view, we need to work with `bodyParser` as well as we cannot restrict our users from using that.

**Comment by bhumilsarvaiya** [comment]
Hey @rajivharlalka, thanks for reporting this bug. The issue is when you are using urlencoded bodyParser, if the request body content-type is of type json, it was not getting parsed. In the library, we are doing parsing if the req.body is either undfined, Buffer or of type string. But because of urlencoded bodyParser, the req.body was getting set to `{}` and thus the sdk was not parsing the request content. I’m working on the fix so that the sdk will always do json parsing on the request and it will be pushed to v8.6.2. But till the fix has been pushed, if you want this to work correctly for now, just add a json body parser.
```js
app.use(bodyParser.json());
```

**Comment by rajivharlalka** [comment]
Sounds Cool, Thanks for the update. 🚀

**Comment by rishabhpoddar** [comment]
Fixed in version 9.0.1
